pkg/standalone-utils/contracts/ProtocolFeeSplitter.sol

```diff
@@ -0,0 +1,130 @@
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/standalone-utils/IProtocolFeeSplitter.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/Authentication.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/VaultHelpers.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IProtocolFeesCollector.sol";
+
+interface Pool {
+    function getOwner() external returns (address);
+}
+
+/**
+ * @dev This contract is responsible for splitting the BPT profits collected
+ * by ProtocolFeeCollector between the pool's owner and Balancers DAO treasury
+ * Nothing happens for non-BPT tokens (WETH, WBTC, etc...)
+ */
+contract ProtocolFeeSplitter is IProtocolFeeSplitter, Authentication {
+    using Math for uint256;
+
+    // All fee percentages are 18-decimal fixed point numbers.
+    // Absolute maximum fee percentages (1e18 = 100%, 1e16 = 1%).
+    uint256 private constant _MAX_REVENUE_SHARING_FEE_PERCENTAGE = 50e16; // 50%
+    address private constant _DELEGATE_OWNER = 0xBA1BA1ba1BA1bA1bA1Ba1BA1ba1BA1bA1ba1ba1B;
+
+    IProtocolFeesCollector public immutable protocolFeesCollector;
+
+    // Balancer DAO Multisig
+    address public immutable treasury;
+
+    // Allows for a pool revenue override
+    mapping(bytes32 => uint256) public revenueSharePerPool;
+
+    constructor(IProtocolFeesCollector _protocolFeesCollector, address _treasury)
+        // The ProtocolFeeSplitter is a singleton, so it simply uses its own address to disambiguate action
+        // identifiers.
+        Authentication(bytes32(uint256(address(this))))
+    {
+        protocolFeesCollector = _protocolFeesCollector;
+        treasury = _treasury;
+    }
+
+    function setRevenueSharingFeePercentage(bytes32 poolId, uint256 newSwapFeePercentage)
+        external
+        override
+        authenticate
+    {
+        _require(newSwapFeePercentage <= _MAX_REVENUE_SHARING_FEE_PERCENTAGE, Errors.SPLITTER_FEE_PERCENTAGE_TOO_HIGH);
+        revenueSharePerPool[poolId] = newSwapFeePercentage;
+        emit PoolRevenueShareChanged(poolId, newSwapFeePercentage);
+    }
+
+    function collectFees(bytes32 poolId) external override {
+        address pool = VaultHelpers.toPoolAddress(poolId);
+        IERC20 bpt = IERC20(pool);
+
+        IProtocolFeesCollector protocolFeesCollectorMemory = protocolFeesCollector;
+        uint256 feeCollectorBptBalance = bpt.balanceOf(address(protocolFeesCollectorMemory));
```

Comment:
Btw, if we are redeeming all of the bpt balance of protocolFeesCollector, does it mean that there is going to be 1 protocol fee splitter for each pool

Reply:
No, ProtocolFeeSplitter is universal.

Pool = BPT (each pool is a different BPT)
FeeCollector collects BPT
FeeSplitter redeems all of BPT from FeeCollector, and splits it between the owner / forwards all to treasury

Reply:
Hmm, then if there is just a single FeeSplitter, why are we redeeming all of the BPT?

Reply:
BC as I understand FeeCollector is a single contract as well, which receives fees from all of the pools, this means that our BPTs should be distributed between 100+ owners, not just a single one

Reply:
If i understood the proposal correctly, only pool owner = pool creator should get the reward. If we wanted to distribute reward to everybody, we would then need to burn the collected BPT, and I don't think that this proposal is about that.

Reply:
But there are few pools connected to the single contract with few pool owners

Reply:
I mean reward gets distributed between the pools and between the pool owners of each of them